### PR TITLE
clean up deploy revisions

### PR DESCRIPTION
**Pull request by Simpcyclassy**
## Description

This PR creates a cron that runs every 23:15 to clean up deploy revisions older than 90 days

## Related Issues

Related  astronomer/issues#6394

## Testing

local testing

## Merging

0.35

### CODE REVIEW

File: charts/astronomer/values.yaml [hunk 1]
```diff
@@ -222,6 +222,18 @@ houston:
...
+  # This runs as a CronJob
+  cleanupDeployRevisions:
+    # Run at 23:40 every night
+    schedule: "40 23 * * *"  # https://crontab.guru/#40_23_*_*_*
+
+    # Cleanup deploy revisions older than this many days
+    olderThan: 90
+
+    # Print out the deployments that should be cleaned up and skip actual cleanup
+    dryRun: false
```
Comment (pgvishnuram) [inline charts/astronomer/values.yaml:235]: we dont support dryRun for  rollback table cleanup

File: charts/astronomer/templates/houston/houston-configmap.yaml [hunk 2]
```diff
@@ -86,6 +86,10 @@ data:
         taskUsageMetricsEnabled: {{ .Values.global.taskUsageMetricsEnabled }}
         taskUsageReportNumberOfDays: {{ .Values.houston.cleanupTaskUsageData.olderThan}}
 
+      deployRollback:
+        enabled: {{ .Values.global.deployRollbackEnabled }}
+        deployRevisionReportNumberOfDays: {{ .Values.houston.cleanupDeployRevisions.olderThan}}
+
```
Comment (pgvishnuram) [inline charts/astronomer/templates/houston/houston-configmap.yaml:92]: enableUpdateDeploymentImageEndpoint this flag also to be included when rollback feature  is enabled

File: charts/astronomer/values.yaml [hunk 3]
```diff
@@ -210,8 +210,8 @@ houston:
   # Cleanup task usage data and task audit data in Houston
   # This runs as a CronJob
   cleanupTaskUsageData:
-    # Run at 23:40 every night
-    schedule: "40 23 * * *"  # https://crontab.guru/#40_23_*_*_*
+    # Run at 23:00 every night
+    schedule: "00 23 * * *"  # https://crontab.guru/#00_23_*_*_*
```
Comment (danielhoherd) [inline charts/astronomer/values.yaml:214]: We should avoid minute 0 with all our cronjobs. Minute 0 is a hot time for job schedules, and adding more to that will make that minute even more resource demanding. Using a different minute will smooth resource demand out over time.

```suggestion
    # Run at 23:11 every night
    schedule: "11 23 * * *"  # https://crontab.guru/#11_23_*_*_*
```

File: tests/chart_tests/test_astronomer_houston_deploy_revisions.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        self, kube_version
+    ):
+        docs = render_chart(
+            kube_version=kube_version,
+            values={
+                "astronomer": {
+                    "houston": {
+                        "cleanupDeployRevisions": {
+                            "enabled": True,
+                            "schedule": "0 23 * * *",
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_deploy_revisions.py:59]: We should test for something different than the default schedule, something that is easily identifiable as being a test value.

```suggestion
                            "schedule": "1 2 3 4 5",
```

File: tests/chart_tests/test_astronomer_houston_deploy_revisions.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            ],
+        )
+
+        assert len(docs) == 1
+        assert docs[0]["kind"] == "CronJob"
+        assert (
+            docs[0]["metadata"]["name"]
+            == "release-name-houston-cleanup-deploy-revisions"
+        )
+        assert docs[0]["spec"]["schedule"] == "0 23 * * *"
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_deploy_revisions.py:75]:
```suggestion
        assert docs[0]["spec"]["schedule"] == "1 2 3 4 5"
```
```suggestion
        assert docs[0]["spec"]["schedule"] == "0 23 * * *"
```

File: charts/astronomer/values.yaml (hunk 3 above)
Comment (danielhoherd) [reply charts/astronomer/values.yaml:214]: We do not appear to have test cases for cleanupTaskUsageData defaults or custom entries.

File: charts/astronomer/templates/houston/cronjobs/houston-cleanup-deploy-revisions-cronjob.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+{{ toYaml (default .Values.global.platformNodePool.affinity .Values.affinity) | indent 12 }}
+          tolerations:
+{{ toYaml (default .Values.global.platformNodePool.tolerations .Values.tolerations) | indent 12 }}
+          restartPolicy: Never
+{{- include "astronomer.imagePullSecrets" . | indent 10 }}
+          containers:
+            - name: cleanup
+              image: {{ template "houston.image" . }}
+              imagePullPolicy: {{ .Values.images.houston.pullPolicy }}
+              # If you supply only args for a Container, the default Entrypoint defined in the Docker image is run with the args that you supplied.
```
Comment (danielhoherd) [inline charts/astronomer/templates/houston/cronjobs/houston-cleanup-deploy-revisions-cronjob.yaml:47]: Unneeded comment.
```suggestion
```

File: charts/astronomer/values.yaml [hunk 7]
```diff
@@ -222,6 +222,18 @@ houston:
...
     canary: false
 
+  # Cleanup task usage data and task audit data in Houston
+  # This runs as a CronJob
+  cleanupDeployRevisions:
+    # Enable cleanup CronJob
+    enabled: true
+
+    # Run at 23:15 every night
+    schedule: "15 23 * * *"  # https://crontab.guru/#15_23_*_*_*
```
[the remark this answers is not in the document]
Comment (danielhoherd) [reply charts/astronomer/values.yaml:232]: Screenshot 2024-06-05 at 10 22 54 AM

https://cronheatmap.com

**Review by danielhoherd — APPROVED** [review]
LGTM

**Pull request merged**